Tests/Packet++Test/Tests/StpTests.cpp

```diff
@@ -42,6 +43,93 @@ PTF_TEST_CASE(StpConfigurationParsingTests)
 	PTF_ASSERT_EQUAL(stpConfLayer->toString(), "Spanning Tree Configuration");
 } // StpConfigurationParsingTests
 
+PTF_TEST_CASE(StpConfigurationCreationTests)
+{
+	timeval time;
+	gettimeofday(&time, NULL);
+
+	READ_FILE_AND_CREATE_PACKET(1, "PacketExamples/StpConf.dat");
+
+	pcpp::Packet stpPacket(&rawPacket1);
+	PTF_ASSERT_TRUE(stpPacket.isPacketOfType(pcpp::STP));
+
+	pcpp::StpConfigurationBPDULayer *stpConfLayerTgt = stpPacket.getLayerOfType<pcpp::StpConfigurationBPDULayer>();
+	PTF_ASSERT_NOT_NULL(stpConfLayerTgt);
+
+	pcpp::StpConfigurationBPDULayer stpConfLayer;
+
+	stpConfLayer.setFlag(0x0);
+	stpConfLayer.setRootId(0x8064001c0e877800);
+	stpConfLayer.setRootPriority(32768);
+	stpConfLayer.setRootSystemIDExtension(100);
+	stpConfLayer.setRootSystemID(pcpp::MacAddress("00:1c:0e:87:78:00"));
+	stpConfLayer.setPathCost(0x4);
+	stpConfLayer.setBridgeId(0x8064001c0e878500);
+	stpConfLayer.setPortId(0x8004);
+	stpConfLayer.setMessageAge(1);
+	stpConfLayer.setMaximumAge(20);
+	stpConfLayer.setTransmissionInterval(2);
+	stpConfLayer.setForwardDelay(15);
+
+	PTF_ASSERT_EQUAL(stpConfLayer.getDataLen(), stpConfLayerTgt->getDataLen());
+	PTF_ASSERT_BUF_COMPARE(stpConfLayer.getData(), stpConfLayerTgt->getData(), stpConfLayer.getDataLen());
+} // StpConfigurationCreationTests
+
+PTF_TEST_CASE(StpConfigurationEditTests)
+{
+	timeval time;
+	gettimeofday(&time, NULL);
+
+	// Read base packet
+	READ_FILE_AND_CREATE_PACKET(1, "PacketExamples/StpConf.dat");
+	pcpp::Packet stpPacket1(&rawPacket1);
+	PTF_ASSERT_TRUE(stpPacket1.isPacketOfType(pcpp::STP));
+
+	pcpp::StpConfigurationBPDULayer *stpConfLayerOrg = stpPacket1.getLayerOfType<pcpp::StpConfigurationBPDULayer>();
```

Comment:
nit: `stpConfLayerOrg` you probably mean "original"? I'd rename to `stpConfLayerOrig` or `stpConfLayerOriginal`

Same for other variables in the file which end with "Org"